x/gov/keeper/msg_server.go

```diff
@@ -26,6 +26,12 @@ var _ types.MsgServer = msgServer{}
 
 func (k msgServer) SubmitProposal(goCtx context.Context, msg *types.MsgSubmitProposal) (*types.MsgSubmitProposalResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	
+	err := k.Keeper.validateInitialDeposit(ctx, msg.InitialDeposit)
```

Comment:
Might want a chain guard as you pointed out.  E.g: if ctx.BlockHeight() >= core.NewMinInitialDepositeHeight.

In general I think  we have to view these types of "blockheight guards" as logical "soft-forks" in our code and then whenever we produce "state breaking updates" via upgrade handlers or genesis import/export we should probably consider removing all of these logical soft-forks and do a major version bump. This will allow us to "GC" the "soft-forks" whenever we know the database is stable and we have a new snapshot ready for people to sync new nodes from :)